addons/product/models/product_pricelist.py

```diff
@@ -93,7 +93,7 @@ def _compute_price_rule_multi(self, products_qty_partner, date=False, uom_id=Fal
     def _compute_price_rule_get_items(self, products_qty_partner, date, uom_id, prod_tmpl_ids, prod_ids, categ_ids):
         self.ensure_one()
         # Load all rules
-        self.env['product.pricelist.item'].flush(['price', 'currency_id', 'company_id'])
+        self.env['product.pricelist.item'].flush(['price', 'date_start', 'date_end', 'currency_id', 'company_id'])
```

Comment:
This confuses me and I still don't know why some fields are chosen and not others. Why not "categ_id", "product_id", "product_tmpl_id", "pricelist_id", "applied_on" or "min_quantity"?

Reply:
Of course. I asked myself that too.

I wanted to have @rco-odoo comment on this before changing too much.

Reply:
I totally agree, the parameters of `flush()` and the query below seem unrelated.
It looks like a mistake happened while rebasing branches.
The principle is to flush what you use in the query below, i.e.,
```suggestion
        self.env['product.pricelist.item'].flush([
            'categ_id', 'product_tmpl_id', 'product_id', 'pricelist_id',
            'date_start', 'date_end', 'applied_on', 'min_quantity',
        ])
```

Reply:
@rco-odoo 
i've integrated your comment